fauxfactory/__init__.py

```diff
@@ -520,25 +521,49 @@ def gen_negative_integer():
     return gen_integer(max_value=max_value)
 
 
-def gen_ipaddr(ip3=False, ipv6=False):
+def gen_ipaddr(ip3=False, ipv6=False, prefix=None):
     """Generates a random IP address.
 
+    You can also specify an IP address prefix if you are interested eg. in
+    local network address generation etc.
+
     :param bool ip3: Whether to generate a 3 or 4 group IP.
     :param bool ipv6: Whether to generate IPv6 or IPv4
+    :param list prefix: A prefix to be used for IPv4 (like [10, 0, 1])
     :returns: An IP address.
     :rtype: str
+    :raises: ``ValueError`` if ``prefix_ipv4`` is too long either to fit into
```

Comment:
If you have an ip3 and you specify the prefix of length 3, you have no randomness.